src/v/pandaproxy/sharded_client_cache.cc

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2022 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#include "pandaproxy/sharded_client_cache.h"
+
+#include "ssx/future-util.h"
+
+using namespace std::chrono_literals;
+
+namespace pandaproxy {
+
+static constexpr auto clean_timer_period = 10s;
+
+ss::future<> sharded_client_cache::start(
+  ss::smp_service_group sg,
+  YAML::Node const& cfg,
+  config::binding<size_t> max_size,
+  config::binding<std::chrono::milliseconds> keep_alive) {
+    _smp_opts = ss::smp_submit_to_options{sg};
+
+    _clean_timer.set_callback([this] {
+        ssx::spawn_with_gate(_gate, [this] {
+            return _cache.invoke_on_all(
+              _smp_opts,
+              [](kafka_client_cache& cache) { cache.clean_stale_clients(); });
+        });
+    });
+
+    _clean_timer.arm_periodic(clean_timer_period);
+
+    // Pass configs by value here or else risk passing the binding
+    // across shards (i.e., fails oncore_debug_verify)
+    return _cache.start(cfg, max_size(), std::move(keep_alive()));
```

Comment:
by not moving the bindings the configuration is no longer live-updatable, right?

what you can do here to resolve this is to pass a lambda, or there is a thing called like sharded_argument, which will run _on the destination core_ and you can pluck out the corresponding binding on the target core.

there are examples of how to do this in application.cc.